lib/api/funnel.js

```diff
@@ -368,6 +369,24 @@ class Funnel {
    * @returns {Promise<Request>}
    */
   async checkRights (request) {
+    
+    // When the Support of Cookie as Jwt is enabled we check if a jwt cookie is present
+    if (global.kuzzle.config.security.supportBrowserCookieAsJwt) {
+      
+      const isCookieOnly = request.input.args.cookieOnly;
+      // When a request id made with CookieOnly set to true
+      // We try to use the jwt cookie if present as jwt token
+      // otherwise check for jwt as input
+      if (isCookieOnly) {
+
+        if (request.input.headers && request.input.headers.cookie) {
+          const cookie = Cookie.parse(request.input.headers.cookie);
```

Comment:
Can this lead to a parsing error? Also since the client want a `cookieOnly` authentification, if the `cookie.jwt` property is not set (l 385) then it should be considered as an error

Reply:
Cookie.parse can only fail when the string given is empty, wich cannot be the case here

Also I dont think if the client want a `cookieOnly` authentication that the this should be considered as an error, since it may be the first time the user makes a request to Kuzzle to authenticate himself and the cookie might not be set at this time

I will made a slight modification to make sure that if  the request is made with `cookieOnly` that only the cookie should be used even if it's not defined, this way I let the funnel handle the case when the token is missing